src/validation/rules/executable_definitions.rs

```diff
@@ -0,0 +1,117 @@
+use super::ValidationRule;
+use crate::static_graphql::query::*;
+use crate::validation::utils::ValidationError;
+use crate::{ast::QueryVisitor, validation::utils::ValidationContext};
+
+/// Executable definitions
+///
+/// A GraphQL document is only valid for execution if all definitions are either
+/// operation or fragment definitions.
+///
+///
+/// See https://spec.graphql.org/draft/#sec-Executable-Definitions
+pub struct ExecutableDefinitions;
+
+impl QueryVisitor<ValidationContext> for ExecutableDefinitions {
+	fn enter_document(&self, _node: &Document, visitor_context: &mut ValidationContext) {
+		for _definition in &_node.definitions {
+			let definition = _definition.to_string();
+
+			fn is_executable_definition(node: &Definition) -> bool {
+				match node {
+					Definition::Operation(_) => true,
+					Definition::Fragment(_) => true,
+				}
+			}
```

Comment:
So my current struggle is this function. I am trying to make this predicate function https://github.com/graphql/graphql-js/blob/3ab82f4981970f50dd3a96e22a11839e4ec5bb7b/src/language/predicates.ts#L24-L31 

But so far doesn't look like it is workin and not so sure how to make it work :)